src/main/java/io/localmotion/batch/notification/ChatBoxNotificationBatch.java

```diff
@@ -0,0 +1,101 @@
+package io.localmotion.batch.notification;
+
+import io.localmotion.chatbox.ChatMessage;
+import io.localmotion.chatbox.IChatboxRepository;
+import io.localmotion.chatbox.notification.ChatBoxNotification;
+import io.localmotion.chatbox.notification.IChatBoxNotificationRepository;
+import io.localmotion.personaldata.IPersonalDataRepository;
+import io.micronaut.scheduling.annotation.Scheduled;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Singleton
+public class ChatBoxNotificationBatch {
+
+	@Inject
+	IChatboxRepository chatboxRepository;
+
+	@Inject
+	IChatBoxNotificationRepository chatBoxNotificationRepository;
+
+	@Inject
+	IPersonalDataRepository personalDataRepository;
+
+
+
+	// job is triggered once a month
+	@Scheduled(cron = "0 0 0 1 * ?")
+	void notifyUnreadMessages() {
+		log.info("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Notification Batch Job Started at: {} ~~~~~~~~~~~~~~~~~~~~~~~~~~~", Instant.now());
+		// TODO: get all distinct user ids from ChatNotification table
+		List<String> persons = chatBoxNotificationRepository.findDistinctPersons();
+		log.info("List of Persons : {}", persons);
+		persons.stream()
+				.forEach( person -> {
+                    PersonNotification personNotification = new PersonNotification();
```

Comment:
Since ChatBoxNotification holds the only userId, we need to get the username and emailId from PersonalDataRecord before we send it to the Email template.